crates/client/rpc/src/lib.rs

```diff
@@ -400,6 +401,67 @@ where
         Ok(block.header().transaction_count)
     }
 
+    /// Gets the Transaction Status, Including Mempool Status and Execution Details
+    ///
+    /// This method retrieves the status of a specified transaction. It provides information on
+    /// whether the transaction is still in the mempool, has been executed, or dropped from the
+    /// mempool. The status includes both finality status and execution status of the
+    /// transaction.
+    ///
+    /// ### Arguments
+    ///
+    /// * `transaction_hash` - The hash of the transaction for which the status is requested.
+    ///
+    /// ### Returns
+    ///
+    /// * `transaction_status` - An object containing the transaction status details:
+    ///   - `finality_status`: The finality status of the transaction, indicating whether it is
+    ///     confirmed, pending, or rejected.
+    ///   - `execution_status`: The execution status of the transaction, providing details on the
+    ///     execution outcome if the transaction has been processed.
+    fn get_transaction_status(&self, transaction_hash: FieldElement) -> RpcResult<TransactionStatus> {
+        let block_hash_from_db = self
+            .backend
+            .mapping()
+            .block_hash_from_transaction_hash(H256::from(transaction_hash.to_bytes_be()))
+            .map_err(|e| {
+                error!("Failed to get transaction's substrate block hash from mapping_db: {e}");
+                StarknetRpcApiError::TxnHashNotFound
+            })?;
+
+        let substrate_block_hash = block_hash_from_db.ok_or(StarknetRpcApiError::TxnHashNotFound)?;
+
+        let block = get_block_by_block_hash(self.client.as_ref(), substrate_block_hash)
+            .ok_or(StarknetRpcApiError::BlockNotFound)?;
+
+        let chain_id = self.chain_id()?.0.into();
+
+        let find_tx = block
+            .transactions()
+            .iter()
+            .find(|tx| tx.compute_hash::<H>(chain_id, false).0 == transaction_hash)
+            .map(|tx| to_starknet_core_tx(tx.clone(), transaction_hash))
+            .ok_or(StarknetRpcApiError::TxnHashNotFound)?;
+
+        let execution_result: TransactionExecutionStatus = self
+            .client
+            .runtime_api()
+            .get_tx_execution_outcome(substrate_block_hash, Felt252Wrapper(transaction_hash).into())
+            .map_err(|e| {
+                error!(
+                    "Failed to get transaction execution outcome. Substrate block hash: {substrate_block_hash}, \
+                     transaction hash: {transaction_hash}, error: {e}"
+                );
+                StarknetRpcApiError::InternalServerError
+            })?
+            .map_or(TransactionExecutionStatus::Succeeded, |message| TransactionExecutionStatus::Reverted);
+
+        Ok(TransactionStatus {
+            finality_status: TransactionFinalityStatus::AcceptedOnL2,
+            execution_status: execution_result,
+        })
+    }
```

Comment:
What about the other finality status?
Recieved (in the memepool)
Rejected (could not be put in the mempool)

Reply:
Okay, but create the required issue with enough background for somebody to jump in